backend/src/services/trading_service.py

```
@@ -33,199 +33,157 @@ def __init__(self, socketio=None):
-            return None
+        for _ in range(max_retries):
+            try:
+                sig = Signature.from_string(signature_str)
+                status = self.solana_client.get_signature_statuses([sig])
+                if status.value and status.value[0]:
+                    if status.value[0].confirmation_status in ['confirmed', 'finalized']:
+                        if status.value[0].err:
+                            logger.error(f"Transaction {signature_str} failed with error: {status.value[0].err}")
+                            return False
+                        return True
+            except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)